woocommerce-pdf-invoices-packingslips.php

```diff
@@ -452,6 +453,51 @@ public function rtl_detected() {
 			}
 		}
 	}
+	
+	/**
+	 * Yearly reset action missing notice
+	 * 
+	 * @return void
+	 */
+	public function yearly_reset_action_missing_notice(): void {
+		if ( ! $this->settings->maybe_schedule_yearly_reset_numbers() ) {
+			return;
+		}
+		
+		if ( ! function_exists( 'as_get_scheduled_actions' ) ) {
+			return;
+		}
+		
+		$current_date   = new \DateTime();
+		$end_of_year    = new \DateTime( 'last day of December' );
+		$days_remaining = $current_date->diff( $end_of_year )->days;
+
+		// Check if the current date is within the last 30 days of the year
+		if ( $days_remaining <= 30 && ! $this->settings->yearly_reset_action_is_scheduled() ) {
+			ob_start();
+			?>
+			<div class="notice notice-error">
+				<p><?php esc_html_e( "The year-end is approaching, and we noticed that your PDF Invoices & Packing Slips for WooCommerce plugin doesn't have the scheduled action to reset invoice numbers annually, even though you've explicitly enabled this setting in the document options. Click the button below to schedule the action before the year ends.", 'woocommerce-pdf-invoices-packing-slips' ); ?></p>
+				<p><a class="button" href="<?php echo esc_url( wp_nonce_url( add_query_arg( 'wpo_wcpdf_schedule_yearly_reset_action', 'true' ), 'schedule_yearly_reset_action_nonce' ) ); ?>"><?php esc_html_e( 'Schedule the action now', 'woocommerce-pdf-invoices-packing-slips' ); ?></a></p>
```

Comment:
Maybe also adding a dimiss button would be good?

Reply:
I prefer not to, as it could be clicked accidentally, either by the admin or a collaborator. Instead, clicking the button should dismiss the notice while still allowing the issue to be resolved.